win32ss/printing/base/winspool/printers.c

```diff
@@ -448,10 +449,20 @@ EnumPrintersA(DWORD Flags, PSTR Name, DWORD Level, PBYTE pPrinterEnum, DWORD cbB
     }
  
     /* Ref: https://stackoverflow.com/questions/41147180/why-enumprintersa-and-enumprintersw-request-the-same-amount-of-memory */
-    bReturnValue = EnumPrintersW(Flags, pwszName, Level, pPrinterEnum, cbBuf, pcbNeeded, pcReturned);
-    HeapFree(hProcessHeap, 0, pwszName);
+    bResult = EnumPrintersW(Flags, pwszName, Level, pPrinterEnum, cbBuf, pcbNeeded, pcReturned);
+    dwErrorCode = GetLastError();
```

Comment:
This is only relevant for the failure case, so please move it into the `if (!bResult)` condition. Follows the usual `if (!returnvalue) GetLastError()` pattern.

Reply:
I mostly concur and would do it if `TRACE()` was (re)moved too.
With `HeapFree()` previously, still with `TRACE()` and possibly with future code in between, I think it is safer to do it like this.

Reply:
Eventually, I added 2 commits, so you can choose from.